src/Sulu/Bundle/AdminBundle/DependencyInjection/Compiler/ExposeResourceRoutesPass.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\AdminBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * This compiler pass is responsible for adding all configured admin resource
+ * routes to `fos_js_routing.routes_to_expose` extension config automatically.
+ */
+class ExposeResourceRoutesPass implements CompilerPassInterface
+{
+    private string $extensionAlias;
+
+    public function __construct(string $extensionAlias)
+    {
+        $this->extensionAlias = $extensionAlias;
+    }
+
+    public function process(ContainerBuilder $container): void
+    {
+        $resources = $container->getParameter($this->extensionAlias . '.resources');
```

Comment:
As beneath no need to use a variable here as it is defined and will not be changed. Make things more readable.

```suggestion
    public function process(ContainerBuilder $container): void
    {
        $resources = $container->getParameter('sulu_admin.resources');
```